File: python/run_all_evals.py
import os
import subprocess
import sys
from pathlib import Path

# Component types to test
component_types = ["plugins", "adapters", "wallets"]

def run_tests_for_component_type(component_type):
    """Run tests for a specific component type"""
    print(f"\n=== Running tests for {component_type} ===\n")
    
    components_path = Path("src") / component_type
    components = [d for d in components_path.iterdir() if d.is_dir()]
    
    for component in components:
        test_file_path = component / "test_eval.py"
        
        if test_file_path.exists():
            print(f"Testing {component.name}...")
            try:
                subprocess.run([sys.executable, str(test_file_path)], 
                              check=True, 
                              capture_output=False)
                print(f"✅ {component.name} tests completed successfully\n")
            except subprocess.CalledProcessError as e:
                print(f"❌ {component.name} tests failed: {e}\n")
        else:
            print(f"⚠️ {component.name} does not have a test file\n")

# Run tests for all component types
for component_type in component_types:
    run_tests_for_component_type(component_type)

print("\n=== All evaluation tests completed ===\n")

File: typescript/run-all-evals.ts
import { execSync } from "node:child_process";
import * as fs from "node:fs";
import * as path from "node:path";

// Component types to test
const componentTypes = ["plugins", "adapters", "wallets"];

// Function to run tests for a component type
function runTestsForComponentType(componentType: string) {
    console.log(`\n=== Running tests for ${componentType} ===\n`);

    const componentsPath = path.join(__dirname, "packages", componentType);
    const components = fs
        .readdirSync(componentsPath)
        .filter((dir) => fs.statSync(path.join(componentsPath, dir)).isDirectory());

    for (const component of components) {
        const testFilePath = path.join(componentsPath, component, "test-eval.ts");

        if (fs.existsSync(testFilePath)) {
            console.log(`Testing ${component}...`);
            try {
                execSync(`ts-node ${testFilePath}`, { stdio: "inherit" });
                console.log(`✅ ${component} tests completed successfully\n`);
            } catch (error) {
                console.error(`❌ ${component} tests failed:`, error);
            }
        } else {
            console.log(`⚠️ ${component} does not have a test file\n`);
        }
    }
}

// Run tests for all component types
for (const componentType of componentTypes) {
    runTestsForComponentType(componentType);
}

console.log("\n=== All evaluation tests completed ===\n");
